app/models/user.rb

```diff
@@ -1,10 +1,33 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  include PgSearch::Model
+
+  attr_accessor :terms_of_service
+
+  has_one_attached :image, dependent: :destroy
+
+  pg_search_scope :search, against: [:id, :first_name, :email]
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable,:confirmable
 
-  attr_accessor :terms_of_service
+  enum role: [:user, :admin]
+  after_initialize :set_default_role, :if => :new_record?
+
+  def set_default_role
+    self.role ||= :user
+  end
+
+  def self.to_csv
+    attributes = %w{ email first_name country phone }
+    CSV.generate(headers: true) do |csv|
+      csv << attributes
+      all.each do |user|
+        csv << attributes.map{ |attr| user.send(attr) }
```

Comment:
Don't use `send` (here it is fine however in public facing actions it is a security risk so we tend to avoid it, if you cannot find any alternative use `public_send`)
```suggestion
        csv << attributes.map{ |attr| user[attr] }
```